backend/src/index.ts

```diff
@@ -0,0 +1,227 @@
+import express from "express";
+import dotenv from "dotenv";
+import generator from "generate-password";
+import { Post, PrismaClient } from "@prisma/client";
+import { createUser, editUser, getAllUsers, getPosts, getUser } from "./db";
+import {
+  PrismaClientKnownRequestError,
+  PrismaClientValidationError,
+} from "@prisma/client/runtime/library";
+import cors from "cors";
+dotenv.config();
+
+const prisma = new PrismaClient();
+
+const getVoteCount = (post) => {
+  const upVotesCount = post.votes.filter(
+    (vote) => vote.voteType === "Upvote"
+  ).length;
+  const downVotesCount = post.votes.length - upVotesCount;
+  return upVotesCount - downVotesCount + 1;
+};
+
+async function main() {
+  await prisma.$connect();
+  console.log("Connected to Prisma");
+}
+
+main()
+  .then(async () => {
+    await prisma.$disconnect();
+  })
+  .catch(async (e) => {
+    console.error(e);
+    await prisma.$disconnect();
+    process.exit(1);
+  });
+
+const app = express();
+app.use(express.json());
+app.use(cors());
+const port = process.env.PORT;
+
+app.get("/", (req: any, res: { send: (arg0: string) => void }) => {
+  res.send("Express + TypeScript Server");
+});
+
+app.get("/users/all", async (req, res) => {
+  const users = await getAllUsers();
+  const userArray = users.map((user) => {
+    return { ...user, password: undefined };
+  });
+
+  res.json(userArray);
+});
+
+app.get("/users", async (req, res) => {
+  const email = req.query;
```

Comment:
```suggestion
  const { email }= req.query;
```
Asaik `req.query` is an object, so the conversion to string might go wrong if more query params are included.